_posts/2020-03-04-arc-limitations.adoc

```diff
@@ -22,52 +22,58 @@ It's been responsible for leading the http://cdi-spec.org/[specification], imple
 
 Our primary goal was to provide a DI solution that would be aligned with the Quarkus mindset:
 
-* Backed by a standard,
+* Based on well known standard,
 * Build-time oriented,
 * Fast and lightweight.
 
-Speaking of standards, the choice was clear: CDI - a technology we have extensive experience at all levels with.
-But very soon we realized that we need a lighter alternative.
+Speaking of standards, the choice was clear: CDI - a technology we have extensive experience with at all levels.
+But we soon realized that the current reference implementation (Weld) won't do -  we needed a lighter alternative.
 
 First, we tried to identify the _dispensable features_ (`D`) that:
 
 * Are rarely used,
 * Cause troubles to users and/or implementors,
+* Do not fit well into the build-time oriented scheme,
 * Make the implementation overly complex, ie. less maintainable and harder to understand.
 
-Afterwards, we compiled a list of _must-have features_ (`M`), such as type-safe resolution, observers and interceptors.
+Afterwards, we compiled a list of _must-have features_ (`M`), such as:
+
+* Type-safe resolution,
+* Observers, interceptors
+* Alternatives,
+* Proxies and much more...
 
 [NOTE]
 ====
-We should mention that the set was built from users point of view.
-For the integration part the Quarkus extension framework should be leveraged.
+It is noteworthy that the set was built from users point of view.
+For the integration part, Quarkus extension framework should be leveraged.
 ====
 
-Then, we idenfitied the resulting set of features `F` as `F = C \ (D \ M)` where `C` is a set of all CDI features.  
+Then we identified the resulting set of features `F` as `F = C \ (D \ M)` where `C` is a set of all CDI features.
```

Comment:
This is probably better done with words then formulas, it took me a while to grasp it and I am still not sure it holds true :)